staging/src/k8s.io/component-helpers/dra/kubeletplugin/draplugin.go

```diff
@@ -45,7 +50,7 @@ type draPlugin struct {
 
 // Start sets up two gRPC servers (one for registration, one for the DRA node
 // client).
-func Start(logger klog.Logger, driverName, endpoint, draAddress, pluginRegistrationPath string, nodeServer drapbv1.NodeServer) (result DRAPlugin, finalErr error) {
+func Start(logger klog.Logger, driverName string, endpoint Endpoint, draAddress string, pluginRegistrationEndpoint Endpoint, nodeServer drapbv1.NodeServer) (result DRAPlugin, finalErr error) {
```

Comment:
This is an API break. We can do that (still early phase of the code), no API stability guarantees for staging repos), but it could become very annoying if this happens again later.

Should this package perhaps use functional parameters? Then it is much easier to support optional parameters (for example, the logger) or support different alternatives (pass just a string for a Unix domain endpoint).

@klueska, @byako: what do you think? I expect you both to use this at some point...

Reply:
You do you mean of the style `WithLogger(logger klog.Logger)`?  Which ones would be required and which ones optional?

Reply:
Yes, although the `With` prefix is not always used: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
I intend to add more options for error injection, which would have been another API change. Let's use functional options.

I've pushed the change as a separate commit in case that you don't agree, but will squash before merging.

Reply:
I like the functional options. But, frankly I'm newbie to Go, so I won't insist on having it.